src/clm/functions.py

```diff
@@ -451,16 +451,67 @@ def get_mass_range(mass, err_ppm):
     return min_mass, max_mass
 
 
-def write_to_csv_file(file_name, df):
-    dirname = os.path.dirname(file_name)
-    if dirname:
-        os.makedirs(dirname, exist_ok=True)
-    df.to_csv(
+def write_to_csv_file(
+    file_name,
+    info,
+    mode="w",
+    header=True,
+    columns=None,
+    string_format=None,
+):
+    # Make an output directory if it doesn't yet
+    os.makedirs(os.path.dirname(file_name), exist_ok=True)
+
+    # See if the provided information is a dataframe
+    if isinstance(info, pd.DataFrame):
+        info.to_csv(
+            file_name,
+            mode=mode,
+            header=header,
+            columns=columns,
+            index=False,
+            compression="gzip" if str(file_name).endswith(".gz") else None,
+        )
+    else:
+        with open(file_name, mode=mode) as f:
+            for row in info:
+                formatted_row = string_format % row
+                f.write(formatted_row)
+
+
+def read_csv_file(
+    file_name,
```

Comment:
the signature can just be `read_csv_file(filename, **kwargs)`, - pass it all on to `pd.read_csv` (adding `compression` as a special case, after popping it off `kwargs` if it also exists there).